doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,428 @@
+used vocabulary.  Other glossary terms are defined throughout the
+manual and the reader may refer to the :ref:`glossary index
+<glossary_index>` for a more complete reference.
+
+Syntax and lexical conventions
+------------------------------
+
+Syntax conventions
+~~~~~~~~~~~~~~~~~~
+
+The grammar of Coq is an extensible, non-context-free grammar.  The
+extensions to the Coq grammar can come from either plugins, or the
+:ref:`notation system <syntaxextensionsandinterpretationscopes>`.
+Furthermore, some parsing rules are only activated in certain contexts
+(interactive proof mode, custom entries...).
+
+Nonetheless, the grammars are presented in this manual as if they were
+context-free.  Precedence levels, when they exist, are represented by
+appending a number to the name of the nonterminal (as in :n:`@term100`
+or :n:`@ltac_expr3`).  Precedence levels which are unused by the base
+grammar are not represented, even though they could still be populated
+by notations or plugins.
+
+In the grammars presented in this manual, the terminal symbols are
+black (e.g. :n:`forall`), whereas the nonterminals are green, italic and
+clickable (e.g. :n:`@term`).  A system of boxes is used to simplify
+some expressions.  Optional expressions are represented as :n:`{? optional }`,
+non-empty, space-separated sequences are represented as :n:`{+ repeated }`,
+possibly empty, space-separated sequences are represented as :n:`{* repeated }`.
+Finally, sequences with separators are represented like so (example
+for the :n:`,` separator): :n:`{+, repeated }`.
+
+.. _lexical-conventions:
+
+Lexical conventions
+~~~~~~~~~~~~~~~~~~~
+
+Blanks
+  Space, newline and horizontal tab are considered blanks.  Blanks are
+  ignored but they separate tokens.
+
+Comments
+  Comments are enclosed between ``(*`` and ``*)``.  They can be
+  nested.  They can contain any character. However, embedded
+  :n:`@string` literals must be correctly closed. Comments are treated
+  as blanks.
+
+Identifiers
+  Identifiers, written :n:`@ident`, are sequences of letters, digits,
+  ``_`` and ``'``, that do not start with a digit or ``'``.  That is,
+  they are recognized by the following grammar (except that the string
+  ``_`` is reserved; it is not a valid identifier):
+
+  .. insertprodn ident subsequent_letter
+
+  .. prodn::
+     ident ::= @first_letter {* @subsequent_letter }
+     first_letter ::= {| a .. z | A .. Z | _ | @unicode_letter }
+     subsequent_letter ::= {| @first_letter | @digit | ' | @unicode_id_part }
+
+  All characters are meaningful. In particular, identifiers are
+  case-sensitive.  :production:`unicode_letter` non-exhaustively
+  includes Latin, Greek, Gothic, Cyrillic, Arabic, Hebrew, Georgian,
+  Hangul, Hiragana and Katakana characters, CJK ideographs,
+  mathematical letter-like symbols and non-breaking
+  space. :production:`unicode_id_part` non-exhaustively includes
+  symbols for prime letters and subscripts.
```

Comment:
Seems strange to me to include subscripts but not superscripts, even in a non-exhaustive list

Reply:
Do you know in which list they should appear? I don't know much about the Unicode support.

Reply:
Pretty sure subscripts and superscripts should appear in the same list, whichever the list that is.  I'd believe Unicode id part, but I'm not sure.

Reply:
I wonder what "prime letters" mean.

Reply:
Perhaps they are referring to prime characters in Unicode, which has characters for single, double triple prime, both forward and backward.

Subscripts work, but superscripts are not accepted:

![image](https://user-images.githubusercontent.com/1253341/80636322-21d6ef80-8a12-11ea-92f9-99e248a6f1a2.png)

Gives `Syntax Error: Lexer: Undefined token`.  And notice the subscripts are not all aligned the same way.  (This is in CoqIde.)

Look at unicode.ml and the very dull unicodetable.ml.

Reply:
Oooh, is this because superscripts are considered to be exponent-like things, where as subscripts are part of identifiers?

Reply:
OK, I can revert this change.